SWIPs/swip-honey_to_money.md

```diff
@@ -1,57 +1,96 @@
+This update process should be atomic: either all nodes upgrade or none of them do. The current implementation hardcodes honey prices in the source code. This violates the atomicity requirement as there is no guarantee that all Swarm nodes are on the same release. By externalizing the price to an oracle, Swarm completely decouples its core functions from price discovery. By building support for price oracles, it will be possible to coordinate price changes across the network and set the stage for future experimentation with various mechanisms for setting prices, from fixed prices to DAOs or to bonding curves.
 
 ## Specification
-* The Swarm source code will reference the address of a `HoneyToMoney` price oracle
-* The price oracle implements the `HoneyToMoney` interface (to-be specified) 
-* The `HoneyToMoney` contract will be initially owned by an M/N multi-sig of Swarm developers and stakeholders.
-* Before sending a payment (denominated in currency), a node will query the `HoneyToMoney` price oracle to get the most up-to-date price conversion. 
-* Upon receiving a payment, nodes will determine the honey amount paid by looking up and dividing by the currently valid price from the oracle. This might cause payment imbalances between nodes, as the oracle might be queried one or several blocks before the transaction was sent. To avoid this situation, we propose that nodes include a payload with the transaction, specifying the time at which the oracle was queried.
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
+* An oracle is reachable at one specific endpoint and this endpoint is hardcoded in the Swarm source-code as a default parameter.
+* If nodes query the oracle at the same time, the oracle gives the same answer.
+* A price oracle returns a `honeyPrice` object, specifying a `Time To Live (TTL)` in seconds and a `price` object. The `price` object maps all `validFrom` entries to their respective `honeyPriceWei`. `validFrom` is a UNIX timestamp and `honeyPriceWei` is the price of one honey unit, denominated in Wei. To summarize, the oracle returns:
+```json
+{
+   "honeyPrices": {
+       "TTL": <TTLValue>,
+       "price": {
+           "validFrom0": <priceInWei0>,
+		…,
+           "validFromN": <priceInWeiN>
+       }
+   }
+}
+```
+An example of this could be:
+```json
+{
+   "honeyPrices": {
+      "TTL": 3600,
+       "price": {
+           "1566396921": 80000,
```

Comment:
Now we see why it is good to have the example: in the description, "validFrom0" suggested it would be a string, but now, in your example, we can actually see that it is a timestamp, a numerical value, thus, it is probably better to not use string, but a numerical value.

Reply:
I see what you mean here and why you are confused. The only reason I made "validFrom" a string is that this is a requirement for json syntax. It does, obviously, not have to be a string. I think it is clear from what it is, but if you think we can improve here, please tell me!

Reply:
I consider this conversation as resolved.

As we agreed on: we are merging this SWIP now. Feel free to raise this as a point of discussion with the wider community!